runtime/src/dispatcher.rs

```diff
@@ -155,16 +160,13 @@ impl Dispatcher {
         info!(self.logger, "Starting the runtime dispatcher");
         let mut rpc_demux = RpcDemux::new(self.rak.clone());
         let mut rpc_dispatcher = RpcDispatcher::new();
-        let mut txn_dispatcher = TxnDispatcher::new();
-        if let Some(initializer) = initializer {
-            initializer.init(
-                &protocol,
-                &self.rak,
-                &mut rpc_demux,
-                &mut rpc_dispatcher,
-                &mut txn_dispatcher,
-            );
-        }
+        let mut txn_dispatcher: Box<dyn TxnDispatcher> = if let Some(txn) =
+            initializer.init(&protocol, &self.rak, &mut rpc_demux, &mut rpc_dispatcher)
+        {
+            txn
+        } else {
+            Box::new(TxnMethDispatcher::new())
```

Comment:
what if we made the initializer do this internally

Reply:
Not sure we can... whoever constructs a concrete dispatcher will need its specific interface, so it would be problematic if the initializer itself returned the general boxed trait already.

Reply:
ok what about yet another implementation that's specifically a no-op dispatcher